dali/operators/image/resize/resize_attr_test.cc

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/resize/resize_attr.h"  // NOLINT
+#include <gtest/gtest.h>
+#include <memory>
+
+namespace dali {
+
+TEST(ResizeAttr, ParseLayout) {
+  auto test = [](const TensorLayout &layout, int expected_spatial_ndim, int expected_first_dim) {
+    int spatial_ndim = -1, first_spatial_dim = -1;
+    EXPECT_NO_THROW(ResizeAttr::ParseLayout(spatial_ndim, first_spatial_dim, layout));
+    EXPECT_EQ(spatial_ndim, expected_spatial_ndim);
+    EXPECT_EQ(first_spatial_dim, expected_first_dim);
+  };
+  test("HW", 2, 0);
+  test("DHW", 3, 0);
+
+  test("HWC", 2, 0);
+  test("CHW", 2, 1);
+
+  test("DHWC", 3, 0);
+  test("CDHW", 3, 1);
+
+  test("FHWC", 2, 1);
+  test("FCHW", 2, 2);
+
+  test("FDHWC", 3, 1);
+  test("FCDHW", 3, 2);
+
+  {
+    int spatial_ndim = -1, first_spatial_dim = -1;
+    EXPECT_THROW(ResizeAttr::ParseLayout(spatial_ndim, first_spatial_dim, "HCW"),
+                 std::runtime_error);
+    EXPECT_THROW(ResizeAttr::ParseLayout(spatial_ndim, first_spatial_dim, "FWCH"),
+                 std::runtime_error);
+  }
```

Comment:
Maybe it should be a separate test case?

Reply:
Can be.